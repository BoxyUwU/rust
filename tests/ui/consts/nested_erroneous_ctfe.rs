fn main() {
    [9; || [9; []]];
    //~^ ERROR: mismatched types
    //~| ERROR: mismatched types
}
