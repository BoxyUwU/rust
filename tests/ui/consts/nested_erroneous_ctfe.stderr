error[E0308]: mismatched types
  --> $DIR/nested_erroneous_ctfe.rs:2:16
   |
LL |     [9; || [9; []]];
   |                ^^ expected `usize`, found `[_; 0]`
   |
   = note: expected type `usize`
             found array `[_; 0]`

error[E0308]: mismatched types
  --> $DIR/nested_erroneous_ctfe.rs:2:9
   |
LL |     [9; || [9; []]];
   |         ^^^^^^^^^^ expected `usize`, found closure
   |
   = note: expected type `usize`
           found closure `[closure@$DIR/nested_erroneous_ctfe.rs:2:9: 2:11]`
help: use parentheses to call this closure
   |
LL |     [9; (|| [9; []])()];
   |         +          +++

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
